src/PrestaShopBundle/Form/Validator/Constraints/TinyMceMaxLengthValidator.php

```diff
@@ -27,25 +27,54 @@
 namespace PrestaShopBundle\Form\Validator\Constraints;
 
 use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Adapter\Validate;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
 
 class TinyMceMaxLengthValidator extends ConstraintValidator
 {
+    /**
+     * @var Validate
+     */
+    private $validateAdapter;
+
+    public function __construct(Validate $validate)
+    {
+        $this->validateAdapter = $validate;
+    }
+
     public function validate($value, Constraint $constraint)
     {
+        if (!$constraint instanceof TinyMceMaxLength) {
+            throw new UnexpectedTypeException($constraint, TinyMceMaxLength::class);
+        }
+
+        if (!$this->validateAdapter->isUnsignedInt($constraint->max)) {
+            throw new \InvalidArgumentException('Max must be int. Input was: ' . \gettype($constraint->max));
+        }
+
         $replaceArray = [
             "\n",
             "\r",
             "\n\r",
         ];
         $str = str_replace($replaceArray, [''], strip_tags($value));
 
-        if ($constraint instanceof TinyMceMaxLength && iconv_strlen($str) > $constraint->max) {
-            $this->context->addViolation(
-                (new LegacyContext())->getContext()->getTranslator()->trans('This value is too long. It should have %limit% characters or less.', [], 'Admin.Catalog.Notification'),
-                ['%limit%' => $constraint->max]
+        if (iconv_strlen($str) > $constraint->max) {
+            $translator = (new LegacyContext())->getContext()->getTranslator();
```

Comment:
Could you move the translator to the constructor ? Better for unit testing.

Reply:
Not related to the PR, but why not :)

Reply:
```suggestion
```

Reply:
done, thanks !

Reply:
(but no pushed :stuck_out_tongue: )

Reply:
still alcohol effect ... very sorry 😔
Done

Reply:
(but no pushed  )

Reply:
😂 thanks ! done for sure now